src/utils/generatePermitHook.ts

```diff
@@ -0,0 +1,97 @@
+import { Erc20Abi } from '@cowprotocol/abis'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Web3Provider } from '@ethersproject/providers'
+import { MaxUint256, Token } from '@uniswap/sdk-core'
+
+import { splitSignature } from 'ethers/lib/utils'
+
+import { GP_VAULT_RELAYER } from 'legacy/constants'
+import { getContract } from 'legacy/utils'
+
+import { toKeccak256 } from '../modules/appData/utils/buildAppData'
+
+export interface PermitHookParams {
+  inputToken: Token
```

Comment:
Why not use `Currency` instead?

```suggestion
  inputToken: Currency
```

Reply:
Because Currency incorporates native, and 1) native doesn't have address and 2) native doesn't need permit.